pkg/controller/acmeorders/sync.go

```diff
@@ -144,10 +148,15 @@ func (c *controller) Sync(ctx context.Context, o *cmacme.Order) (err error) {
 		return err
 	}
 
+	acmeOrder, err := getACMEOrder(ctx, cl, o)
+	if err != nil {
+		return err
+	}
```

Comment:
Having just read the #3938 PR, I wondered if there are any permanent ACME errors which we should be checking for here? What if the ACME server has deleted the order at its end? Won't we get a 404 here and then keep retrying for ever?

Reply:
See https://github.com/jetstack/cert-manager/pull/3805/files#diff-59361284baf54294f984ac681e4885dec8fb7e3afa476819d704bb2703b4f179R165-R171 below, where we cast to acmeapi.Error and check the status code is between 400 and 500 before setting the order to failed.